lib/internal/quic/session.js

```diff
@@ -0,0 +1,1188 @@
+        certificate,
+        issuer,
+        callback,
+      };
+    },
+    [],
+    OCSPRequest);
+}
+
+function createOCSPResponse(response) {
+  return ReflectConstruct(
+    function() {
+      this[kType] = 'OCSPResponse';
+      this[kState] = { response };
+    },
+    [],
+    OCSPResponse);
+}
+
+/**
+ * @param {string} alpn
+ * @param {string} servername
+ * @param {string[]} ciphers
+ * @param {ClientHelloCallback} callback
+ * @returns {void}
+ */
+function createClientHello(alpn, servername, ciphers, callback) {
+  return ReflectConstruct(
+    function() {
+      this[kType] = 'ClientHello';
+      this[kState] = {
+        alpn,
+        servername,
+        ciphers,
+        callback,
+      };
+    },
+    [],
+    ClientHello);
+}
+
+/**
+ * @typedef {import('../abort_controller').AbortSignal} AbortSignal
+ * @param {EndpointWrap} handle
+ * @returns {Session}
+ */
+function createSession(
+  handle,
+  endpoint,
+  alpn = undefined,
+  authority = undefined) {
+  const ret = ReflectConstruct(
+    class extends NodeEventTarget {
+      constructor() {
+        super();
+        this[kType] = 'Session';
+        const inner = new SessionState(handle.state);
+        this[kState] = {
+          alpn,
+          application: selectApplication(alpn),
+          authority,
+          certificate: undefined,
+          closeRequested: false,
+          closed: createDeferredPromise(),
+          cipher: undefined,
+          clienthello: undefined,
+          earlyData: false,
+          endpoint,
+          handshake: createDeferredPromise(),
+          inner,
+          keylog: undefined,
+          maxPacketLength: 0,
+          ocsp: undefined,
+          peerCertificate: undefined,
+          qlog: undefined,
+          remoteAddress: undefined,
+          servername: undefined,
+          sessionTicket: undefined,
+          silentClose: false,
+          statelessReset: false,
+          streams: new SafeMap(),
+          validationError: undefined,
+        };
+        this[kHandle] = handle;
+        this[kStats] = createSessionStats(handle.stats);
+        setPromiseHandled(this[kState].closed.promise);
+        setPromiseHandled(this[kState].handshake.promise);
+
+        if (inner.clientHello) {
+          this[kState].clienthello = createDeferredPromise();
+          setPromiseHandled(this[kState].clienthello.promise);
+        }
+
+        if (inner.ocsp) {
+          this[kState].ocsp = createDeferredPromise();
+          setPromiseHandled(this[kState].ocsp.promise);
+        }
+      }
+    },
+    [],
+    Session);
+  ret[kHandle][owner_symbol] = ret;
+  return ret;
+}
+
+function isSessionDestroyed(session) {
+  return session[kHandle] === undefined;
+}
+
+function sessionRemoteAddress(session) {
+  if (session[kState].remoteAddress === undefined) {
+    const ret = session[kHandle]?.getRemoteAddress();
+    session[kState].remoteAddress =
+      ret !== undefined ? new InternalSocketAddress(ret) : undefined;
```

Comment:
```suggestion
      ret === undefined ? undefined : new InternalSocketAddress(ret);
```
I think it's easier to read.